lib/engine/game/g_18_uruguay/game.rb

```diff
@@ -0,0 +1,908 @@
+        }.freeze
+
+        MARKET_TEXT = Base::MARKET_TEXT.merge(
+          par: 'Par value',
+          repar: 'RLTP stock price',
+          close: 'Close',
+          endgame: 'End game trigger',
+        )
+        STOCKMARKET_COLORS = Base::STOCKMARKET_COLORS.merge(
+          liquidation: :darkred
+        )
+        EVENTS_TEXT = Base::EVENTS_TEXT.merge(
+          nationalization: ['Nationalization',
+                            'Time for nationalization']
+        ).freeze
+
+        def price_movement_chart
+          [
+            ['Action', 'Share Price Change'],
+            ['Dividend 0 or withheld', '1 ←'],
+            ['Dividend paid', '1 →'],
+            ['One or more shares sold (Except RLTP)', '1 ↓'],
+            ['Corporation sold out at end of SR', '1 ↑'],
+          ]
+        end
+
+        def init_optional_rules(optional_rules)
+          super
+        end
+
+        def find_and_remove_train_by_id(train_id)
+          train = train_by_id(train_id)
+          return unless train
+
+          @depot.remove_train(train)
+          train.buyable = true
+          train.reserved = true
+          train
+        end
+
+        def setup
+          super
+          @rptla = @corporations.find { |c| c.id == 'RPTLA' }
+          @fce = @corporations.find { |c| c.id == 'FCE' }
+
+          @rptla.add_ability(Engine::G18Uruguay::Ability::Ship.new(
+            type: 'Ship',
+            description: 'Ship goods'
+          ))
+          @rptla.add_ability(Engine::Ability::Base.new(
+            type: 'Goods',
+            description: GOODS_DESCRIPTION_STR + '0',
+            count: 0
+          ))
+          @rptla.add_ability(Engine::Ability::Base.new(
+            type: 'Incremental',
+            description: 'Incremental capatilization',
+            count: 0
+          ))
+          @stock_market.set_par(@rptla, lookup_rptla_price(RPTLA_STARTING_PRICE))
+
+          ability = abilities(corn_farm, :assign_hexes, time: 'or_start', strict_time: false)
+          ability.hexes.each do |farm_id|
+            hex_by_id(farm_id).assign!('GOODS_CORN')
+          end
+
+          ability = abilities(sheep_farm, :assign_hexes, time: 'or_start', strict_time: false)
+          ability.hexes.each do |farm_id|
+            hex_by_id(farm_id).assign!('GOODS_SHEEP')
+          end
+
+          ability = abilities(cattle_farm, :assign_hexes, time: 'or_start', strict_time: false)
+          ability.hexes.each do |farm_id|
+            hex_by_id(farm_id).assign!('GOODS_CATTLE')
+          end
+
+          setup_destinations
+        end
+
+        def init_train_handler
+          depot = super
+
+          # Due to that the nationalization happens on second 6 purchase an not on a phase change,
+          # the event needs to be tied to a train.
+          train = depot.upcoming.reverse.find { |t| t.name == '6' }
+          train.events << { 'type' => 'nationalization' }
```

Comment:
This can be done via config in trains.rb, with `when => 2`, see 18NY - https://github.com/tobymao/18xx/blob/06307a9a6e5702aa2a07bd4c020fb10d753d7ca0/lib/engine/game/g_18_ny/game.rb#L147